Objects/obmalloc.c

```diff
@@ -139,7 +139,10 @@ _PyMem_mi_page_maybe_free(mi_page_t *page, mi_page_queue_t *pq, bool force)
 
         _PyMem_mi_page_clear_qsbr(page);
         page->retire_expire = 0;
-        page->qsbr_goal = _Py_qsbr_deferred_advance(tstate->qsbr);
+
+        size_t bsize = mi_page_block_size(page);
+        page->qsbr_goal = _Py_qsbr_advance_with_size(tstate->qsbr, page->capacity*bsize);
```

Comment:
This might be the right heuristic, but this is a bit different from `_PyMem_FreeDelayed`:

1) `_PyMem_FreeDelayed` holds onto the memory until quiescence. It prevents the memory from being used for any purpose.
2) `_PyMem_mi_page_maybe_free` only prevents the page from being used by another thread or for a different size class. That's a lot less restrictive.

Reply:
Ah good point.  The memory being held (avoiding collection) by mimalloc is not at all the same as the deferred frees.   I reworked the PR so that memory is tracked separately.  I also decoupled the write sequence advance from the triggering of `_PyMem_ProcessDelayed()`, used `process_seq` as a target value for the read sequence.  

Now `_qsbr_thread_state` is larger than 64 bytes.  I don't think that should be a problem.